### ISSUE TITLE
Facial Recognition doesn't work

### ISSUE THREAD

**Issue by Mert5** [paviro/MMM-Facial-Recognition#56]
When I download zip and add the code to the config.js, I continuously see the same error. I tried to change the folder that the training.xml inside, also in the config.js but the error is the same. I enabled camera from sudo raspi-config, removed the camera module that you talked in another topic because the probability of interfere, but nothing changed. Just one more thing, Once the module worked but I could not catch that again(at that time I tried to change some script and I added the clock while the module working and achieved but changing much things caused lose of working module). Sorry for the long explanation. Any help is remarkable.

**Comment by Mert5** [comment]
The code that I add config is:
{
			module: "MMM-Facial-Recognition",
			position: "bottom",
			config: {
				// 1=LBPH | 2=Fisher | 3=Eigen
				recognitionAlgorithm: 1,
				// Threshold for the confidence of a recognized face before it's considered a
				// positive match. Confidence values below this threshold will be considered
				// a positive match because the lower the confidence value, or distance, the
				// more confident the algorithm is that the face was correctly detected.
				IbphThreshold: 50,
				fisherThreshold: 250,
				eigenThreshold: 3000,
				// force the use of a usb webcam on raspberry pi (on other platforms this is  always true automatically)
				useUSBCam: true,
				// Path to your training xml
				trainingFile: "modules/MMM-Facial-Recognition/training.xml",
				// recognition intervall in seconds (smaller number = faster but CPU intens!)
				interval: 2,
				// Logout delay after last recognition so that a user does not get instantly logged out
				// if he turns away from the mirror for a few seconds
				logoutDelay: 15,
				// Array with usernames (copy and paste from training script)
				users: ['Mert'],
				// Module set used for strangers and if no user is detected
				defaultClass: "default",
				// Set of modules which should be shown for every user
				everyoneClass: "everyone",
				// Boolean to toggle welcomeMessage
				welcomeMessage: true
				}
		},
		{
			module: "example_module2",
			position: "top_right",
			//Set your classes here seperated by a space.
			//Shown for all users
			classes: "Mert",
		},

**Comment by Mert5** [comment]
And the error that I continuously see:
Starting MagicMirror: v2.3.1
Loading config ...
Loading module helpers ...
Initializing new module helper ...
Module helper loaded: MMM-Facial-Recognition
No helper found for module: example_module2.
All module helpers loaded.
Starting server on port 8080 ... 
Server started ...
Connecting socket for: MMM-Facial-Recognition
Starting module helper: MMM-Facial-Recognition
Sockets connected & modules started ...
Launching application.
[MMM-Facial-Recognition] Facerecognition started...
[MMM-Facial-Recognition] Loading training data...
[MMM-Facial-Recognition] ALGORITHM: LBPH
[MMM-Facial-Recognition] Training data loaded!
[MMM-Facial-Recognition] --------------------
[MMM-Facial-Recognition] Webcam ausgewÃ¤hlt...
Whoops! There was an uncaught exception...
{ # #***_*Error: HIGHGUI ERROR: V4L: index 0 is not correct!**_**
**_Traceback (most recent call last):
  File "modules/MMM-Facial-Recognition/facerecognition/facerecognition.py", line 63, in <module>
    camera = config.get_camera()
  File "/home/pi/MagicMirror/modules/MMM-Facial-Recognition/facerecognition/config.py", line 57, in get_camera
    return webcam.OpenCVCapture(device_id=0)
  File "/home/pi/MagicMirror/modules/MMM-Facial-Recognition/facerecognition/webcam.py", line 31, in __init__
    self._camera.open()
TypeError: Required argument 'device' (pos 1) not found_**

    at PythonShell.parseError (/home/pi/MagicMirror/modules/MMM-Facial-Recognition/node_modules/python-shell/index.js:191:17)
    at terminateIfNeeded (/home/pi/MagicMirror/modules/MMM-Facial-Recognition/node_modules/python-shell/index.js:98:28)
    at ChildProcess.<anonymous> (/home/pi/MagicMirror/modules/MMM-Facial-Recognition/node_modules/python-shell/index.js:89:9)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
  executable: 'python',
  options: null,
  script: 'modules/MMM-Facial-Recognition/facerecognition/facerecognition.py',
  args: [ '{"recognitionAlgorithm":1,"lbphThreshold":50,"fisherThreshold":250,"eigenThreshold":3000,"useUSBCam":true,"trainingFile":"modules/MMM-Facial-Recognition/training.xml","interval":2,"logoutDelay":15,"users":["Mert"],"defaultClass":"default","everyoneClass":"everyone","welcomeMessage":true,"IbphThreshold":50}' ],
  exitCode: 1 }
MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
Shutting down server...
Stopping module helper: MMM-Facial-Recognition

**Comment by Mert5** [comment]
I guess I solved the problem.
I did not do so much different things as I told above. Now I again have "Hello stranger,.." alert. Sometimes you just need to wait for a while not more than 1-2 minutes to see this alert. If you took the same error as I told before, there are three possibilities of this error:
1-Internet speed,
2-Background light(if it is too shine, it can give this error)
3-The camera(I am using an old webcam, so this can also be a problem-but the same webcam sometimes give the alert as I told above)
I hope I can help someone, I am closing this issue.
